src/schema/v2/me/artistRecommendations.ts

```diff
@@ -0,0 +1,84 @@
+import { GraphQLFieldConfig, GraphQLInt } from "graphql"
+import { connectionFromArraySlice } from "graphql-relay"
+import { convertConnectionArgsToGravityArgs } from "lib/helpers"
+import { CursorPageable, pageable } from "relay-cursor-paging"
+import { createPageCursors } from "schema/v1/fields/pagination"
+import { ResolverContext } from "types/graphql"
+import gql from "lib/gql"
+import { artistConnection } from "schema/v2/artist"
+import { find } from "lodash"
+
+const MAX_ARTISTS = 50
+
+export const ArtistRecommendations: GraphQLFieldConfig<
+  void,
+  ResolverContext
+> = {
+  description: "A connection of artist recommendations for the current user.",
+  type: artistConnection.connectionType,
+  args: pageable({
+    page: { type: GraphQLInt },
+  }),
+  resolve: async (
+    _root,
+    args: CursorPageable,
+    { vortexGraphqlLoader, artistsLoader }
+  ) => {
+    if (!vortexGraphqlLoader || !artistsLoader) return
+
+    const { page, size, offset } = convertConnectionArgsToGravityArgs(args)
+
+    // Fetch artist IDs from Vortex
+
+    const vortexResult = await vortexGraphqlLoader({
+      query: gql`
+        query artistRecommendationsQuery {
+          artistRecommendations(first: ${MAX_ARTISTS}) {
+            totalCount
+            edges {
+              node {
+                artistId
+                score
+              }
+            }
+          }
+        }
+      `,
+    })()
+
+    const artistIds: string[] = vortexResult.data?.artistRecommendations?.edges?.map(
+      (edge) => edge?.node?.artistId
+    )
+
+    // Fetch artist details from Gravity
+
+    let artists: any = []
+
+    if (artistIds?.length) {
+      artists = (
+        await artistsLoader({
+          ids: artistIds,
+          size,
+          offset,
+        })
+      ).body
+
+      // Apply order from Vortex result (score ASC)
+
+      artists = artistIds.map((artistId) => {
```

Comment:
```suggestion
      artists = artistIds.filter((artistId) => {
```
**issue:** Shouldn't this be `filter` 🤔. In the case where `artistIds` will initially have 50 recommended artists and we have queried gravity for only `size = 10` artists, artists will be an array containing `undefined` values coming from `find`

Reply:
Good catch!

We actually don't have to pass `offset` and `size` to **Gravity**. The number of artist IDs coming from **Vortex** will be enough. 

I added `compact` to remove all `undefined` values in the case an artist can't be found in **Gravity**.
```
artists = compact(
        artistIds.map((artistId) => {
          return find(artists, { _id: artistId })
        })
      )
```

Reply:
aha. thanks 🙌

Reply:
I can't find the compact in your last commit 🤔

Reply:
Forgot to push and added it now 🙇